### PR TITLE
[CARBONDATA-2800][Doc] Add useful tips about bloomfilter datamap

### PR DESCRIPTION
**Pull request by xuchuanyin**
add useful tips about bloomfilter datamap

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 `NO`
 - [x] Any backward compatibility impacted?
  `NO`
 - [x] Document update required?
`Yes, updated the document`
 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
`NA`
        - How it is tested? Please attach test report.
`NA`
        - Is it a performance related change? Please attach the performance test report.
`NA`
        - Any additional information to help reviewers in testing this change.
`NA`
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7620/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7649/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6375/

File: docs/useful-tips-on-carbondata.md [hunk 1]
```diff
@@ -125,6 +125,10 @@
     TBLPROPERTIES ('SORT_COLUMNS'='Dime_1, HOST, MSISDN')
   ```
 
+  **NOTE:**
+  + BloomFilter can be created to enhance performance for queries with precise equal/in conditions. You can find more information about it in BloomFilter datamap document.
+
```
Comment (chenliang613) [inline docs/useful-tips-on-carbondata.md]: please add one link to bloomfilter datamap document.

File: docs/datamap/bloomfilter-datamap-guide.md [hunk 2]
```diff
@@ -103,3 +104,24 @@ If the datamap does not prune blocklets well, you can try to increase the value
...
++ We can create multiple BloomFilter datamaps on one table,
+ also we can create one BloomFilter datamap that contains multiple index columns.
+ We do recommend the later behavior since the data loading and query performance will be better.
++ `BLOOM_FPP` is only the expected number from user, the actually FPP may be worse.
+ If the BloomFilter datamap does not work well,
+ you can try to increase `BLOOM_SIZE` and decrease `BLOOM_FPP` at the same time.
+ Notice that bigger `BLOOM_SIZE` will increase the size of index file
+ and smaller `BLOOM_FPP` will increase runtime calculation while performing query.
++ '0' skipped blocklets of BloomFilter datamap in explain output indicates that
+ BloomFilter datamap does not prune better than Main datamap.
```
Comment (chenliang613) [inline docs/datamap/bloomfilter-datamap-guide.md]: BloomFilter datamap does not prune better than Main datamap.  -- can you provide more detail about this point.

File: docs/useful-tips-on-carbondata.md (hunk 1 above)
Comment (xuchuanyin) [reply docs/useful-tips-on-carbondata.md]: OK

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/6075/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7668/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7701/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/6094/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**